son-gtklic/licenseManager/views/licenses.py

```diff
@@ -0,0 +1,173 @@
+        if license.expiringDate < datetime.now():
+            return build_response(status_code=400, data=license.serialize, error="License is not valid")
+
+        return build_response(status_code=200, data=license.serialize, description="License is valid")
+
+    def post(self, licenseID):
+        try:
+            license = License.query.get(licenseID)
+            if license is None:
+                return build_response(status_code=404, error="Not Found", description="License ID provided does not exist")
+
+            if 'user_uuid' not in request.form:
+                return build_response(status_code=400, error="Missing Field", description="user_uuid was not provided")
+
+            if license.user_uuid != request.form['user_uuid']:
+                return build_response(status_code=400, error="Invalid User", description="user_uuid provided is not the one that owns the license")
+
+            license_type = Type.query.get(license.type_uuid)
+            if 'type_uuid' in request.form:
+                license_type = Type.query.get(request.form['type_uuid'])
+                if not license_type:
+                    return build_response(status_code=404, error="Invalid License Type", description="License type provided to change does not exist")
+                license.type_uuid = license_type.type_uuid
+
+            license_service = Service.query.get(license.service_uuid)
+
+            baseExpiringDate = license.expiringDate
+            if license.expiringDate < datetime.now():
+                baseExpiringDate = datetime.now()
+
+            new_date = baseExpiringDate + timedelta(days=license_type.duration)
+
+            if new_date > license_service.expiringDate:
+                return build_response(status_code=400, error="Invalid date", description="Service no longer available for the license period")
+            license.expiringDate = new_date
+
+        except:
+            return build_response(status_code=400, error="Invalid arguments", description="Arguments provided were invalid")
+
+        db.session.commit()
+
+        return build_response(status_code=200, data=license.serialize, description="License successfully renewed")
+
+    def put(self, licenseID):
+        license = License.query.filter_by(license_uuid=licenseID).first()
+        if license is None:
+            return build_response(status_code=404, error="Not Found", description="License ID provided does not exist")
+
+        if not license.active:
+            return build_response(status_code=400, error="Not Valid", description="License ID provided is cancelled")
+
+        if license.suspended:
+            return build_response(status_code=304, error="Already Suspended", description="License ID provided is already suspended")
+
+        license.suspended = True
+        db.session.commit()
+
+        return build_response(status_code=200, data=license.serialize, description="License successfully suspended")
+
+    def delete(self, licenseID):
```

Comment:
We'll have to revisit 'delete', when we define what should be the end-to-end behaviour when there are service instances running when delete is invoked